packages/react-components/react-timepicker-compat-preview/src/components/TimePicker/useTimePicker.tsx

```diff
@@ -66,24 +69,40 @@ export const useTimePicker_unstable = (props: TimePickerProps, ref: React.Ref<HT
     initialState: undefined,
   });
 
+  const selectedTimeTextRef = React.useRef<string | undefined>(undefined);
+
+  const selectTime: TimePickerProps['onTimeSelect'] = React.useCallback(
+    (e, data) => {
+      setSelectedTime(data.selectedTime);
+      selectedTimeTextRef.current = data.selectedTimeText;
```

Comment:
`selectedTimeTextRef` is only ever set with `setSelectedItem`, does it need to be a ref?

Also isn't it more like `selectedTimeTextUncommitted` or `selectedTimeTextRaw`?

Reply:
Do you mean `let selectedTimeTextUncommitted` instead of ref? I think that'd also work.

Reply:
Synced on call and update to use state